README.md
affiliates-manager-plugin
=========================

Official github repository of the Affiliates Manager WordPress plugin


wpaffiliatemanager/changelog.txt
//////////////////////////////////////////////////

                WP Affiliate Manager
                  Affiliate Plugin
                     ChangeLog

//////////////////////////////////////////////////

-- Version 1.6.5, Updated: 3/21/14 --

- Trim creative slug to prevent spaces in Easy Digital Downloads affiliate links


-- Version 1.6.4, Updated: 1/6/14 --

- Fixed javascript error on custom registration field edit
- Fixed invalid refkey error for certain affiliates during refkey unpack()
- Fixed percentage of sale on Easy Digital Downloads sales


-- Version 1.6.3, Updated: 11/2/13 --

- Fixed automatic (licensed) updates


-- Version 1.6.2, Updated: 8/30/13 --

- Fixed login redirect conflicts with other plugins (Peter's Login Redirect)
- Added preliminary (Google Translate) translations for: Bulgarian, Hebrew, German, Portugese, Dutch, Finnish, French, Spanish and Italian


-- Version 1.6.1, Updated: 7/8/13 --

- iThemes Exchange Support
- Added 'wpam_credit_amount' filter to dynamically change affiliate payout amount
- Changed cookie.js to actual file instead of symlink to avoid problems with zip decompressors that do not support symlinks


-- Version 1.6, Updated: 6/27/13 --

- Easy Digital Downloads Support
- WooCommerce Support
- Custom Affiliate Link URLs
- i18n ready


-- Version 1.5.1, Updated: 2/22/13 --

- Fixed resolveConflict() from 1.5.0 to *correctly* rename tables wpsca -> wpam
- Reworked new affiliate to be able to add existing WP users from botched wpsca -> wpam rename (1.5.0)
- Simplified email validation


-- Version 1.5.0, Updated: 12/26/12 --

- Moved WPSCA options/capabilities/pages/tables namespace to WPAM (fixes issue w/ registration errors on new WPEC 3.8.9.x installations)
- Added translation wrappers to text fields (minus paypal & settings admin pages)
- Added money_format function for Windows
- Fixed issue of submitting to site_url instead of home_url for registration (only affected sub-directory installs)
- Removed curl requirement for MassPay
- Updated PayPal sandbox URL
- Fixed display of affiliate links containing quotes
- Use core WP jquery scripts except for cookies & style (jquery ui theme)


-- Version 1.4.4, Updated: 3/16/12 --

- Warn if problems loading locale other than en_US (and force to USD)
- Load front-end shortcode javascript in the footer to avoid wpautop() formatting issues
- Made ajax calls go through admin-ajax.php for maximum compatibility
- Added debugging for wpsca_refkey problems
- Fixed issue where wpsca_refkey might not be recognised
- Removed mbstring requirement for currency/locale other than en_US


-- Version 1.4.3, Updated: 11/16/11 --

- Added currency flexibility based on WPLANG locale
- Changes to resolve issues with siliconapp theme


-- Version 1.4.2, Updated: 10/27/11 --

- Reworded email change info dialog on admin panel to make sense in context
- Added cookie path to setcookie() to insure cookies get saved in webkit (chrome, safari) browsers


-- Version 1.4.1, Updated: 9/9/11 --

- Changed front end affiliate pages to use standard shortcodes - alleviates issues with certain themes
- Added Affiliate Links tab to My Affiliates page for admin


-- Version 1.4, Updated: 7/27/11 --

- Added screen for admins to manually create an affiliate account
- Added running balance to 'My Affiliates' transaction view for admin
- Added name & email fields to customize emails sent from the Affiliate Manager
- Combined the Affiliate Manager approval email & standard WP login info email into one
- Added the ability to sort 'My Affiliates'
- Added transaction views for affiliates with a date selection: Payment History & Sales
- Added profile screen for affiliates where they can edit their contact information
- Allow affiliate managers to edit their affiliates' contact information and payment details including their bounty type & amount
- Added date selection to PayPal Mass Pay screen
- Changed Mass Pay screen to highlight records that are inactive, and deselect items that are less than minimum payment amount
- Added upload field to new/edit creatives screen
- Changed plugin icon
- Add ability to activate/deactivate affiliates
- Fixed creative being deactivated after edit
- Made email always required
- Updated "How to Setup PayPal Mass Pay.pdf" to reflect new changes in the Affiliate Manager and at PayPal


-- Version 1.3.4, Updated: 7/7/11 --

- Fixed check for WP >= 3.1 on plugin activation


-- Version 1.3.3, Updated: 7/5/11 --

- Added check for WP 3.1 or greater
- Changed cookies to string instead of binary to avoid browser incompatibilities (resets cookies with duration > 0)
- Fixed possibility of affiliateId of zero (0) being returned (will instead show error)
- Fixed 'Closed Transactions' stats from being recorded twice for one purchase


-- Version 1.3.2, Updated: 6/27/11 --

- Replaced php short tags
- Reworked (and removed some) custom jquery to work the 'WordPress' way
- Made Contact Information editable under 'My Affiliates'


-- Version 1.3.1, Updated: 2/28/11 --

- Fixed javascript incompatibility with recently upgraded jquery or WordPress installations
- Added cookie length to affiliate settings
- Added login_redirect filter to send affiliates that need to confirm to the affiliate front-end page
- Fixed affiliate sign-up link for existing WP users to bring them right to the registration page


-- Version 1.3, Updated: 1/23/11 --

Note: if you are upgrading to version 1.3 from a previous version, be sure to read the readme.txt file for information on how this update will affect your existing affiliates

- Added Paypal Mass Pay support for affiliate payouts
- Made system messaging editable via the WP admin dashboard
- Added ability to add custom fields to affiliate registration form
- Fixed issue with manual payout amount being stored incorrectly

wpaffiliatemanager/html/contact_info.php
<?php

require_once WPAM_BASE_DIRECTORY . "/html/widget_form_errors_panel.php";

if( ! isset( $model ) && isset( $this->viewData['affiliate'] ) )
	$model = $this->viewData['affiliate'];

?>
<div id="email_help" style="display: none;">
<p>
<?php
if( is_admin() ):
	_e( "Changing the affiliate's email address will not change their WordPress Affiliate Manager login, which may be based on the email address they used when they signed up.", 'wpam' );
else:
	_e( 'Changing your email address will not change your WordPress Affiliate Manager login, which may be based on the email address you used when you signed up.', 'wpam' );
endif;
?>
</p>
</div>


	<form method="post" id="infoForm">
		<table class="widefat">
			<thead>
				<tr>
					<th colspan="2"><?php echo isset( $this->viewData['infoLabel'] ) ? $this->viewData['infoLabel'] : __( 'Contact Information', 'wpam' ); ?></th>
				</tr>
			</thead>
			<tbody>
			<?php
				foreach ($this->viewData['affiliateFields'] as $field):
					$value = '';
					//prefer request data
					if ( isset( $this->viewData['request']['_' . $field->databaseField] ) ) {
						$value = $this->viewData['request']['_' . $field->databaseField];
					} elseif ( isset( $model ) ) {
						if( $field->type == 'base' )
							$value = $model->{$field->databaseField};
						else
							$value =  $model->userData[$field->databaseField];
					} 
				?>
				<tr>
					<td><label for="_<?php echo $field->databaseField; ?>">
						<?php echo $field->name; ?>
						</label>
						<?php
						echo $field->required ? '&nbsp;*': '';
						if( is_admin() )
							echo $field->type == 'custom' ? '&nbsp;+': '';
						if ( $field->fieldType == 'email' ): ?>
							<img id="emailInfo" style="cursor: pointer;" src="<?php echo WPAM_URL . "/images/info_icon.png"?>" />
						<?php endif; ?>
					</td>
					<td>
						<?php switch ($field->fieldType) {
							case 'string':
							case 'email':
							case 'number':
							case 'zipCode':
								?>
							<input type="text" size="20" id="_<?php echo $field->databaseField; ?>" name="_<?php echo $field->databaseField; ?>" value="<?php echo $value; ?>" />
							<?php break;
							case 'phoneNumber':?>							
							<input type="text" size="20" maxlength="14" id="_<?php echo $field->databaseField; ?>" name="_<?php echo $field->databaseField; ?>" value="<?php echo $value; ?>" />
							<?php break;
							case 'stateCode':?>
							<select id="_<?php echo $field->databaseField; ?>" name="_<?php echo $field->databaseField; ?>">
							<?php wpam_html_state_code_options($value); ?>
							</select>
							<?php break;
							case 'countryCode':?>
							<select id="_<?php echo $field->databaseField; ?>" name="_<?php echo $field->databaseField; ?>">
							<?php wpam_html_country_code_options($value); ?>
							</select>
							<?php break; default: break;
						}?>
					</td>
				</tr>
				<?php if( !is_admin() && $field->fieldType == 'email' && isset( $this->viewData['newEmail'] ) ): ?>
					<tr><td colspan="2"><?php printf( __('There is a pending change of your e-mail to <code>%1$s</code>. <a href="%2$s">Cancel</a>', 'wpam' ), $this->viewData['newEmail']['newemail'], esc_url( self_admin_url( 'profile.php?dismiss=' . $this->viewData['userId'] . '_new_email' ) ) ); ?></td></tr>
				<?php endif; ?>
			<?php endforeach;?>
			</tbody>
			<?php if ( ! isset( $model ) || ( ! $model->isPending() && ! $model->isBlocked() && ! $model->isDeclined() ) ): ?>
			<thead>
				<tr>
					<th colspan="2"><?php _e( 'Payment Details', 'wpam' ) ?></th>
				</tr>
			</thead>
			<tbody>
		<?php if ( isset( $this->viewData['paymentMethods'] ) ): ?>
		<tr>
				<td><label for="ddPaymentMethod"><?php _e( 'Method', 'wpam' ) ?></label> *</td>
				<td><select id="ddPaymentMethod" name="ddPaymentMethod">
					<?php foreach ($this->viewData['paymentMethods'] as $key => $val) {
						$selected_html = $this->viewData['paymentMethod'] == $key ? ' selected="selected"' : '';			
						echo "<option value='{$key}'{$selected_html}>{$val}</option>";
					}?>
				</select></td>
		</tr>
		<tr id="rowPaypalEmail"<?php if( $this->viewData['paymentMethod'] != 'paypal' ) echo ' style="display: none;"'; ?>>
			<td><label for="txtPaypalEmail"><?php _e( 'PayPal E-Mail Address', 'wpam' ) ?></label> *</td>
			<td>
				<input id="txtPaypalEmail" type="text" name="txtPaypalEmail" size="30" value="<?php echo $this->viewData['paypalEmail']?>"/>
			</td>
		</tr>
		<?php endif; ?>
		<?php if ( is_admin() ): ?>					  
		<tr>
			<td>
				<label for="ddBountyType"><?php _e( 'Bounty Type *', 'wpam' ) ?></label>
			</td>
			<td>
				<select id="ddBountyType" name="ddBountyType">
		<?php

					$select = array( 'percent' => __( 'Percentage of Sales', 'wpam' ),
									 'fixed' => __( 'Fixed Amount per Sale', 'wpam' ) );

		$selected = isset( $this->viewData['bountyType'] ) ? $this->viewData['bountyType'] : NULL;
		foreach ( $select as $value => $name ) {
			$selected_html = $value == $selected ? ' selected="selected"' : '';
			echo "<option value='{$value}'{$selected_html}>{$name}</option>\n";
		}

		$currency = WPAM_MoneyHelper::getDollarSign();

		$label = isset( $this->viewData['bountyType'] ) && $this->viewData['bountyType'] == 'fixed' ?
			sprintf( __( 'Bounty Rate (%s per Sale) *', 'wpam' ), $currency) : __( 'Bounty Rate (% of Sale) *', 'wpam' );

		$bountyAmount = isset( $this->viewData['bountyAmount'] ) ? $this->viewData['bountyAmount'] : '';

		?>
				</select>
			</td>
		</tr>
		<tr>
			<td><label id='lblBountyAmount' for='txtBountyAmount'><?php echo $label; ?></label></td>
			<td><input type='text' id='txtBountyAmount' name='txtBountyAmount' size='5' value='<?php echo $bountyAmount; ?>'/></td>
		</tr>
		<?php endif; //is_admin ?>
		<?php endif; //not pending, blocked or declined ?>
			</tbody>
		</table>
		<p><?php _e( '* Required fields', 'wpam' ) ?></p>
		<?php if( is_admin() ): ?>
			<p><?php _e( '+ Custom fields', 'wpam' ) ?></p>
		<?php endif; ?>
		<div class="buttonsBar">			
			<input type="hidden" name="action" value="saveInfo"/>
                        <input type="submit" id="saveInfoButton" class="button-primary" name="wpam-add-affiliate" value="<?php echo isset( $this->viewData['saveLabel'] ) ? $this->viewData['saveLabel'] : __( 'Save Changes', 'wpam' ); ?>" />
		</div>
	    </form>
